CMakeLists.txt

```diff
@@ -72,9 +72,11 @@ add_definitions(-DGHC_WIN_DISABLE_AUTO_PREFIXES)
 
 # Dependencies
 # ============
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH}" "${CMAKE_CURRENT_SOURCE_DIR}/cmake/")
 
 find_package(Threads REQUIRED)
 find_package(nlohmann_json REQUIRED)
+find_package(Sodium REQUIRED)
```

Comment:
maybe we should move this behind a config flag until the feature is ready to be rolled out? So that we can keep the builds more lightweight until we're ready. Doesn't have to happen in this PR, but maybe before the next release.

Reply:
ok!